### PR TITLE
Add FXIOS-5916 [v114] Nimbus onboarding framework setup

### PR DESCRIPTION
**Pull request by adudenamedruby**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-5916)

### Description
The basics of the onboarding framework.

### Pull requests checks where applicable
- [x] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [x] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by adudenamedruby** [comment]
Not ready for review yet.

File: Client/Frontend/Onboarding/Models/OnboardingViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+struct OnboardingViewModel {
+    let cards: [OnboardingCardInfoModel]?
+    let dismissable: Bool
```
Comment (lmarceau) [inline Client/Frontend/Onboarding/Models/OnboardingViewModel.swift:9]: nit, would indicate this is a Boolean with this name better I think?
```suggestion
    let isDismissable: Bool
```

**Comment by adudenamedruby** [comment]
Ready for review.

File: Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    /// Will sort onboarding cards according to specified order in the
+    /// Nimbus configuration. If the names of cards and the names in the card
+    /// order array don't match, these cards will simply not be shown in onboarding.
+    ///
+    /// - Parameters:
+    ///   - cardData: Card data from ``FxNimbus/shared``
+    ///   - cardOrder: Card order from ``FxNimbus/shared``
+    /// - Returns: Card data coverted to ``OnboardingCardInfoModel`` and ordered.
```
Comment (lmarceau) [inline Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift:29]: nit
```suggestion
    /// - Returns: Card data converted to ``OnboardingCardInfoModel`` and ordered.
```

File: Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    /// used by the application in the privacy policy link.
+    private func getOnboardingLink(from cardLink: NimbusOnboardingLink?) -> OnboardingLinkInfoModel? {
+        guard let cardLink = cardLink,
+              let url = URL(string: cardLink.url)
+        else { return nil }
+
+        return OnboardingLinkInfoModel(title: cardLink.title, url: url)
+    }
+
+    /// Translates a nimbus image ID for onboarding to a an ``ImageIdentifiers`` based id
```
Comment (lmarceau) [inline Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift:88]: nit
```suggestion
    /// Translates a nimbus image ID for onboarding to an ``ImageIdentifiers`` based id
```

File: Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    /// that corresponds to an app resource.
+    ///
+    /// In the case that a unknown image identifier in entered into experimenter, the
```
Comment (lmarceau) [inline Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift:91]: nit, I think that's what you meant?
```suggestion
    /// In the case that an unknown image identifier is entered into experimenter, the
```

**Review by lmarceau — COMMENTED** [review]
Some nits, looks great 🎉

File: Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+        image: String,
+        type: String
+    ) -> String {
+        return """
+  "name": "\(CardElementNames.name) \(number)",
+  "title": "\(CardElementNames.title) \(number)",
+  "body": "\(CardElementNames.body) \(number)",
+  "image": "\(image)",
+  "type": "\(type)",
+"""
```
[the remark this answers is not in the document]
Comment (adudenamedruby) [reply Tests/ClientTests/OnboardingTests/NimbusOnboardingFeatureLayerTests.swift:467]: correct.

**Comment by adudenamedruby** [comment]
Suggestions were comment or minor code related that wouldn't change test results or build results.
[Build was green here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/5c112e3d-58a7-4410-875e-6952aefaf1f1). Unless further suggestions require code changes, I will no re-run BR if approved.

**Review by yoanarios — APPROVED** [review]

**Pull request merged**